lib/SILOptimizer/Differentiation/PullbackEmitter.cpp

```diff
@@ -178,6 +178,21 @@ Optional<TangentSpace> PullbackEmitter::getTangentSpace(CanType type) {
       LookUpConformanceInModule(getModule().getSwiftModule()));
 }
 
+SILValueCategory PullbackEmitter::getTangentValueCategory(SILValue v) {
+  // Quick check: if the value has an address type, the tangent value category
```

Comment:
This is not always true. Address only original types can totally have a address-only tangent, simplest example:
```swift
struct Foo<T>: Differentiable {
    var a: Float
    @noDerivative var b: T
}
```

Could you add a FIXME here with an appropriate bug number?